### PR TITLE
sDAI onboarding - Review

### PR DESCRIPTION
**Pull request by marczeller**

### CODE REVIEW

File: src/20230817_AaveV3_Eth_SDAIOnboarding/AaveV3_Ethereum_SDAIOnboarding_20230817.sol [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        optimalStableToTotalDebtRatio: _bpsToRay(20_00)
+      }),
+      enabledToBorrow: EngineFlags.DISABLED,
+      stableRateModeEnabled: EngineFlags.DISABLED,
+      borrowableInIsolation: EngineFlags.DISABLED,
+      withSiloedBorrowing: EngineFlags.DISABLED,
+      flashloanable: EngineFlags.DISABLED,
+      ltv: 77_00,
+      liqThreshold: 80_00,
+      liqBonus: 4_00,
```
Comment (sendra) [inline src/20230817_AaveV3_Eth_SDAIOnboarding/AaveV3_Ethereum_SDAIOnboarding_20230817.sol:43]: Shouldn't this be 4.5 (taken from forum params)

File: src/20230817_AaveV3_Eth_SDAIOnboarding/AaveV3_Ethereum_SDAIOnboarding_20230817.sol [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      reserveFactor: 20_00,
+      supplyCap: 300_000_000,
```
Comment (sendra) [inline src/20230817_AaveV3_Eth_SDAIOnboarding/AaveV3_Ethereum_SDAIOnboarding_20230817.sol:45]: This was not mentioned in the forum post. Does it come from some other conversiation somewhere?

File: diffs/preTestAaveV2_Ethereum_DAIParamsUpdates_20230817_postTestAaveV2_Ethereum_DAIParamsUpdates_20230817.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+| description | value before | value after |
+| --- | --- | --- |
+| reserveFactor | 15 % | 25 % |
+| stableBorrowRateEnabled | true | false |
+| interestRateStrategy | [0xfffE32106A68aA3eD39CcCE673B646423EEaB62a](https://etherscan.io/address/0xfffE32106A68aA3eD39CcCE673B646423EEaB62a) | [0xc6A068E321C83FEacc25C80118E2B1208c54B6ce](https://etherscan.io/address/0xc6A068E321C83FEacc25C80118E2B1208c54B6ce) |
+| optimalUsageRatio | 80 % | 90 % |
+| maxExcessUsageRatio | 20 % | 10 % |
+| variableRateSlope1 | 4 % | 5 % |
+| interestRate | ![before](/.assets/c415a8c57745a62d63e11134fe8acf5059377542.svg) | ![after](/.assets/d1a493171bfb16a17ecaf7a2d7ef8e0782317be5.svg) |
```
Comment (sendra) [inline diffs/preTestAaveV2_Ethereum_DAIParamsUpdates_20230817_postTestAaveV2_Ethereum_DAIParamsUpdates_20230817.md:15]: for some reason looks like these assets don't exist. Maybe forgot to push them? (Maybe need to re run tests to generate again and push new assets?)

File: src/20230817_AaveV3_Eth_SDAIOnboarding/AaveV3_Ethereum_SDAIOnboarding_20230817.sol [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+/**
+ * @title sDAI onboarding
+ * @author Marc Zeller (@marczeller - Aave Chan Initiative)
+ * - Snapshot: https://snapshot.org/#/aave.eth/proposal/0xb41600e3f74bfa520549106fee4d21e7f674c5a79fbf1f7ca16947563474a4d7
+ * - Discussion: https://governance.aave.com/t/arfc-sdai-aave-v3-onboarding/14410
+ */
+contract AaveV3_Ethereum_SDAIOnboarding_20230817 is AaveV3PayloadEthereum {
+  address public constant SDAI = 0x83F20F44975D03b1b09e64809B757c47f942BEeA;
+  address public constant SDAI_PRICE_FEED = 0xb9E6DBFa4De19CCed908BcbFe1d015190678AB5f;
```
Comment (sendra) [inline src/20230817_AaveV3_Eth_SDAIOnboarding/AaveV3_Ethereum_SDAIOnboarding_20230817.sol:16]: Looks like this oracle, uses the dai chainlink feed and then multiplies it with the savings rate of sDai.
Can you double check its correct @rustboyar (as you have more exp with this)

File: src/20230817_AaveV3_Eth_SDAIOnboarding/AaveV3_Ethereum_SDAIOnboarding_20230817.t.sol [hunk 5]
```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import 'forge-std/Test.sol';
```
Comment (sendra) [inline src/20230817_AaveV3_Eth_SDAIOnboarding/AaveV3_Ethereum_SDAIOnboarding_20230817.t.sol:4]: Not needed as its already decalred on ProtocolV3TestBase

File: src/20230817_AaveV3_Eth_SDAIOnboarding/AaveV3_Ethereum_SDAIOnboarding_20230817.t.sol [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+      AaveV3Ethereum.POOL
+    );
+
+    GovHelpers.executePayload(vm, address(proposal), AaveGovernanceV2.SHORT_EXECUTOR);
+
+    ReserveConfig[] memory allConfigsAfter = createConfigurationSnapshot(
+      'postAaveV3_Ethereum_SDAIOnboarding_20230817',
+      AaveV3Ethereum.POOL
+    );
+
```
Comment (sendra) [inline src/20230817_AaveV3_Eth_SDAIOnboarding/AaveV3_Ethereum_SDAIOnboarding_20230817.t.sol:35]: would also put e2e test for sDAI (using underlying address as its not added yet on address-book)

File: diffs/preTestAaveV2_Ethereum_DAIParamsUpdates_20230817_postTestAaveV2_Ethereum_DAIParamsUpdates_20230817.md (hunk 3 above)
Comment (marczeller) [reply diffs/preTestAaveV2_Ethereum_DAIParamsUpdates_20230817_postTestAaveV2_Ethereum_DAIParamsUpdates_20230817.md:15]: assets working correctly locally can't reproduce issue.

File: src/20230817_AaveV3_Eth_SDAIOnboarding/AaveV3_Ethereum_SDAIOnboarding_20230817.t.sol (hunk 5 above)
Comment (marczeller) [reply src/20230817_AaveV3_Eth_SDAIOnboarding/AaveV3_Ethereum_SDAIOnboarding_20230817.t.sol]: fixed

File: src/20230817_AaveV3_Eth_SDAIOnboarding/AaveV3_Ethereum_SDAIOnboarding_20230817.t.sol (hunk 6 above)
Comment (marczeller) [reply src/20230817_AaveV3_Eth_SDAIOnboarding/AaveV3_Ethereum_SDAIOnboarding_20230817.t.sol:35]: Done

File: src/20230817_AaveV3_Eth_SDAIOnboarding/AaveV3_Ethereum_SDAIOnboarding_20230817.sol (hunk 1 above)
Comment (marczeller) [reply src/20230817_AaveV3_Eth_SDAIOnboarding/AaveV3_Ethereum_SDAIOnboarding_20230817.sol]: good catch, fixed

File: src/20230817_AaveV3_Eth_SDAIOnboarding/AaveV3_Ethereum_SDAIOnboarding_20230817.sol (hunk 2 above)
Comment (marczeller) [reply src/20230817_AaveV3_Eth_SDAIOnboarding/AaveV3_Ethereum_SDAIOnboarding_20230817.sol:45]: current value a placeholder, syncing with both risk teams for a consensus value on this.

Comment (marczeller) [reply src/20230817_AaveV3_Eth_SDAIOnboarding/AaveV3_Ethereum_SDAIOnboarding_20230817.sol]: value changed for supply cap from Risk provider recommendation

**Comment by Zer0dot** [comment]
Noticing this also updates the WETH reserve with little mention about it, is that intentional (or did I miss something)?

**Comment by Zer0dot** [comment]
No further comments here, LGTM!

**Review by Zer0dot — DISMISSED** [review]

**Pull request merged**